JDL validation should catch different clientFrameworks when using microfrontends

##### **Overview of the issue**

If you create a JDL that has different client frameworks in microservices, you won't know this fails until runtime. For example, if you modified [reactive-mf.jdl](https://github.com/jhipster/jdl-samples/blob/main/reactive-mf.jdl) to have something like the following:

```
application {
  config {
    ...
    clientFramework angular
    ...
  }
  ...
}

application {
  config {
    ...
    clientFramework react
    ...
  }
  ...
}

application {
  config {
    ...
    clientFramework vue
    ...
  }
  ...
}

...
```

If you generate apps with the above JDL, the gateway will have an error in the browser console and menu items aren't shown.

```
navbar.component.ts:73 Error loading store entities ScriptExternalLoadError: Loading script failed.
(error: http://localhost:8080/services/store/remoteEntry.js)
    at Object.908 (:8080/main.js:46:25)
    at __webpack_require__ (:8080/main.js:83:41)
    at initExternal (:8080/main.js:382:28)
    at Function.__webpack_require__.I (:8080/main.js:411:15)
    at :8080/main.js:543:47
    at Object.3037 (:8080/main.js:597:26)
    at :8080/main.js:670:56
    at Array.forEach (<anonymous>)
    at Object.__webpack_require__.f.consumes (:8080/main.js:653:36)
    at :8080/main.js:157:40
navbar.component.ts:62 Error loading blog entities Error: Module "./entity-navbar-items" does not exist in container.
while loading "./entity-navbar-items" from 1527
    at :8080/services/blog/remoteEntry.js:26:11
    at ZoneDelegate.invoke (:8080/polyfills.js:434:30)
    at Object.onInvoke (:8080/node_modules_angular_core_fesm2015_core_mjs.js:25904:33)
    at ZoneDelegate.invoke (:8080/polyfills.js:433:56)
    at Zone.run (:8080/polyfills.js:194:47)
    at :8080/polyfills.js:1352:38
    at ZoneDelegate.invokeTask (:8080/polyfills.js:468:35)
    at Object.onInvokeTask (:8080/node_modules_angular_core_fesm2015_core_mjs.js:25891:33)
    at ZoneDelegate.invokeTask (:8080/polyfills.js:467:64)
    at Zone.runTask (:8080/polyfills.js:239:51)
```

##### **Motivation for or Use Case**

We should catch issues we know about at generation time rather than runtime.

--- a/cli/import-jdl.js
+++ b/cli/import-jdl.js
@@ -426,12 +426,14 @@ class JDLProcessor {
     );
 
     applicationsWithEntities.forEach((applicationWithEntities, idx) => {
+      const microfrontends = applicationWithEntities.config.microfrontends || [];
+
       const relatedApplications = Object.entries(allApplications).filter(
         ([baseName]) =>
           applicationWithEntities.config.baseName !== baseName &&
-          applicationWithEntities.entities.find(entity => entity.microserviceName === baseName)
+          (applicationWithEntities.entities.find(entity => entity.microserviceName === baseName) || microfrontends.includes(baseName))
       );
-      const { serverPort: gatewayServerPort } = applicationWithEntities.config;
+      const { serverPort: gatewayServerPort, clientFramework: gatewayClientFramework } = applicationWithEntities.config;
       if (relatedApplications.length > 0) {
         applicationWithEntities.config.applications = Object.fromEntries(
           relatedApplications.map(([baseName, config]) => {
@@ -440,6 +442,18 @@ class JDLProcessor {
             return [baseName, { clientFramework, serverPort, applicationIndex, devServerPort }];
           })
         );
+
+        const differentClientFrameworks = relatedApplications.filter(
+          ([_baseName, { clientFramework }]) => clientFramework && clientFramework !== 'no' && clientFramework !== gatewayClientFramework
+        );
+
+        if (differentClientFrameworks.length > 0) {
+          throw Error(
+            `Using different client frameworks in microfrontends is not supported. Tried to use: ${gatewayClientFramework} with ${differentClientFrameworks
+              .map(([baseName, { clientFramework }]) => `${clientFramework} (${baseName})`)
+              .join(', ')}`
+          );
+        }
       }
     });
 

--- a/cli/import-jdl.js
+++ b/cli/import-jdl.js
@@ -425,7 +425,7 @@ class JDLProcessor {
       })
     );
 
-    applicationsWithEntities.forEach((applicationWithEntities, idx) => {
+    applicationsWithEntities.forEach(applicationWithEntities => {
       const microfrontends = applicationWithEntities.config.microfrontends || [];
 
       const relatedApplications = Object.entries(allApplications).filter(

